integration/networking/port_mapping_linux_test.go

```diff
@@ -400,3 +400,43 @@ func TestAccessPublishedPortFromRemoteHost(t *testing.T) {
 		}
 	}
 }
+
+func TestRestartUserlandProxyUnder2MSL(t *testing.T) {
+	skip.If(t, testEnv.IsRootless())
+
+	ctx := setupTest(t)
+
+	d := daemon.New(t)
+	d.StartWithBusybox(ctx, t)
+	defer d.Stop(t)
+
+	c := d.NewClientT(t)
+	defer c.Close()
+
+	const netName = "nat-time-wait"
+	network.CreateNoError(ctx, t, c, netName,
+		network.WithDriver("bridge"),
+		network.WithOption(bridge.BridgeName, netName))
+	defer network.RemoveNoError(ctx, t, c, netName)
+
+	ctrOpts := []func(*container.TestContainerConfig){
+		container.WithName(sanitizeCtrName(t.Name() + "-server")),
+		container.WithExposedPorts("80/tcp"),
+		container.WithPortMap(nat.PortMap{"80/tcp": {{HostPort: "1780"}}}),
+		container.WithCmd("httpd", "-f"),
+		container.WithNetworkMode(netName),
+	}
+
+	serverID := container.Run(ctx, t, c, ctrOpts...)
+	defer c.ContainerRemove(ctx, serverID, containertypes.RemoveOptions{Force: true})
+
+	httpClient := &http.Client{Timeout: 3 * time.Second}
+	resp, err := httpClient.Get("http://127.0.0.1:1780")
+	assert.NilError(t, err)
+	assert.Check(t, is.Equal(resp.StatusCode, 404))
+
+	assert.NilError(t, c.ContainerRemove(ctx, serverID, containertypes.RemoveOptions{Force: true}))
+
+	serverID = container.Run(ctx, t, c, ctrOpts...)
+	defer c.ContainerRemove(ctx, serverID, containertypes.RemoveOptions{Force: true})
```

Comment:
This one doesn't have to be a `defer`;

```suggestion
	assert.NilError(t, c.ContainerRemove(ctx, serverID, containertypes.RemoveOptions{Force: true}))
```

Reply:
Right, actually it doesn't even need to be removed. At this point, the 1st defer will take of deleting that container (if the container name is passed).